File: abc/abc291/a/solution_2.cpp
// https://atcoder.jp/contests/abc291/submissions/65772642
#include <bits/stdc++.h>
using namespace std;

int main() {
    string S;
    cin >> S;
    int N = S.size();

    for (int i = 0; i < N; i++) {
        if ('A' <= S[i] && S[i] <= 'Z') {
            cout << i + 1 << "\n";
        }
    }
}

File: abc/abc326/c/solution_2.py
# https://atcoder.jp/contests/abc326/submissions/65807493
N, M = map(int, input().split())
A = list(map(int, input().split()))
A.sort()

ans = 0
r = 0
# A[l] に左端を投げると [l, r) 番目のプレゼントを捕まえられる
for l in range(N):
    while r < N and A[r] < A[l] + M:
        r += 1
    ans = max(ans, r - l)

print(ans)
